homeassistant/components/home_connect/switch.py

```diff
@@ -261,9 +258,25 @@ async def async_turn_on(self, **kwargs: Any) -> None:
 
     async def async_turn_off(self, **kwargs: Any) -> None:
         """Switch the device off."""
+        if not hasattr(self, "power_off_state"):
+            await self.async_fetch_power_off_state()
+            if not hasattr(self, "power_off_state"):
+                raise ServiceValidationError(
+                    translation_domain=DOMAIN,
+                    translation_key="unable_to_retrieve_turn_off",
+                    translation_placeholders={
+                        SVE_TRANSLATION_KEY_APPLIANCE_NAME: self.device.appliance.name
+                    },
+                )
```

Comment:
Why don't we execute this when the entity gets added to hass? `async_added_to_hass`